### PR TITLE
sync common makefile and scripts with eks-anywhere-build-tooling

### PR DESCRIPTION
**Pull request by jaxesn**
*Issue #, if available:*

*Description of changes:*

This is syncing the common scripts from eks-anywhere-build-tooling over.  The biggest changes are:

- remove create-binaries scripts and use standard go build with configurable flags, this trigggered changes to most checksums because we changed the buildid from "" to nil...
- remove image targets by defining IMAGE_NAMES, recently done in eks-a
- standardize copy/release artifacts into common.mk like in eks-a.  These scripts and calls do not directly align with eks-a but were made as close as possible.

Notes:

- This does bring in some additional targets from eks-a that we arent using, like the cgo support.  I think its easier to keep these in sync if we just pull all this over even if not needed.
- Broken into two commits, the first are the common script changes probably review this the closest.  The second is all the project makefile/script changes.  Github will struggle to render the second commit...

I have tested the postsubmit for these changes using the minimal branch both for a minimal build and a standard build.  I feel pretty good about the changes but will watch the postsubmit once merged to be sure.

/unhold

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kschumy** [comment]
What's the difference between this one and this other PR you have open? https://github.com/aws/eks-distro/pull/633

File: docs/development/add-new-project.md [hunk 1]
```diff
@@ -171,32 +144,34 @@ $(call IMAGE_TARGETS_FOR_NAME, kube-proxy-base): <new_pre_preq>
...
 
 ### patching
 
-Kubernetes is the only project which currently requires patches, but to support that the following was added to the Makefile.
+Kubernetes is the only project which currently requires patches. To support these, the Common.mk will automatically add the following
+if a `patches` directory exists.
 
 ```
-GIT_PATCH_TARGET=$(REPO)/eks-distro-patched
+GIT_PATCH_TARGET=$(REPO)/eks-anywhere-patched
```
Comment (abhay-krishna) [inline docs/development/add-new-project.md:164]: nit
```suggestion
GIT_PATCH_TARGET=$(REPO)/eks-distro-patched
```

**Review by abhay-krishna — APPROVED** [review]

**Pull request merged**